runtime/src/main/java/io/micronaut/jackson/codec/JacksonMediaTypeCodec.java

```diff
@@ -172,7 +173,9 @@ public boolean supportsType(Class<?> type) {
      */
     public <T> T decode(Argument<T> type, JsonNode node) throws CodecException {
         try {
-            return getObjectMapper().treeToValue(node, type.getType());
+            ObjectMapper objectMapper = getObjectMapper();
+            JsonParser jsonParser = objectMapper.treeAsTokens(node);
+            return objectMapper.readValue(jsonParser, constructJavaType(type));
```

Comment:
Please wrap this in a `if (type.hasTypeVariables()) {` as is done above to limit the potential for different behavior

Reply:
sorry, missed your review, done